util/cachedigest/digest.go

```diff
@@ -0,0 +1,159 @@
+package cachedigest
+
+import (
+	"bytes"
+	"encoding/binary"
+	"hash"
+	"regexp"
+	"sync"
+
+	"github.com/moby/buildkit/util/bklog"
+	digest "github.com/opencontainers/go-digest"
+)
+
+type Type string
+
+const (
+	TypeJSON        Type = "json"
+	TypeString      Type = "string"
+	TypeStringArray Type = "string-array"
+	TypeDigestArray Type = "digest-array"
+	TypeFileList    Type = "file-list"
```

Comment:
Can we be consistent? Maybe `-list` instead of array?

```suggestion
	TypeStringList Type = "string-list"
	TypeDigestList Type = "digest-list"
	TypeFileList   Type = "file-list"
```

Reply:
I can change this but logically these are different. File-list is a special format for file headers and checksum per file, not delimiter-separated.